examples/run/run.cpp

```diff
@@ -638,38 +737,40 @@ class LlamaData {
         return 0;
     }
 
-    int resolve_model(std::string & model_) {
-        int                            ret     = 0;
-        if (string_starts_with(model_, "file://") || std::filesystem::exists(model_)) {
+    int resolve_model(std::string & model_, std::string & chat_template_) {
+        int ret = 0;
+        if (string_starts_with(model_, "file://")) {
```

Comment:
Could we re-add "|| std::filesystem::exists(model_)" ?

The logic is supposed to be, if the file just exists, use it otherwise if we don't have a protocol specified, assume ollama

Reply:
I moved this into the download functions to check for the model and template file individually. If the model or template file is there, we'll skip.